### PR TITLE
Correct unbound assertions variable error in shell2junit

### PR DESCRIPTION
**Pull request by cblecker**
**What this PR does / why we need it**:
Fixes the following error in the verify job log:
```
W0126 21:47:20.783] hack/make-rules/../../third_party/forked/shell2junit/sh2ju.sh: line 176: assertions: unbound variable
```

Also adds myself as an OWNER to this little forked mess. If I break it, I bought it lol.

**Release note**:
```release-note
NONE
```

/assign ixdy stevekuznetsov

### CODE REVIEW

File: third_party/forked/shell2junit/sh2ju.sh [hunk 1]
```diff
@@ -46,13 +46,13 @@ errfile=/tmp/evErr.$$.log
 function eVal() {
   (eval "$1")
   # stdout and stderr may currently be inverted (see below) so echo may write to stderr
-  echo $? 2>&1 | tr -d "\n" > $errfile
+  echo $? 2>&1 | tr -d "\n" > ${errfile}
```
Comment (stevekuznetsov) [inline third_party/forked/shell2junit/sh2ju.sh]: If you're doing this you might as well quote it too :)

File: third_party/forked/shell2junit/sh2ju.sh [hunk 2]
```diff
@@ -76,108 +76,108 @@ function juLog() {
   done
 
   # create output directory
-  mkdir -p "$juDIR" || exit
+  mkdir -p "${juDIR}" || exit
   # use first arg as name if it was not given
-  if [ -z "$name" ]; then
-    name="$asserts-$1"
+  if [ -z "${name}" ]; then
```
Comment (stevekuznetsov) [inline third_party/forked/shell2junit/sh2ju.sh]: Similarly, `[` is deprecated so we could just use `[[`

**Review by stevekuznetsov — APPROVED** [review]
Variable defaulting 

/lgtm
/hold

in case anyone cares about the nits

**Comment by cblecker** [comment]
@stevekuznetsov addressed nits. PTAL!

**Pull request merged**